File: backend/app/schemas/menu.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime

class MenuItemBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    description: Optional[str] = Field(None, max_length=500)
    price: float = Field(..., gt=0)
    category: str = Field(..., min_length=1, max_length=50)

class MenuItemCreate(MenuItemBase):
    pass

class MenuItemUpdate(MenuItemBase):
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    price: Optional[float] = Field(None, gt=0)
    category: Optional[str] = Field(None, min_length=1, max_length=50)

class MenuItemInDB(MenuItemBase):
    id: str
    menu_id: str

    class Config:
        from_attributes = True

class MenuItem(MenuItemInDB):
    pass

class MenuBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)

class MenuCreate(MenuBase):
    items: Optional[List[MenuItemCreate]] = None

class MenuUpdate(MenuBase):
    name: Optional[str] = Field(None, min_length=1, max_length=100)

class MenuInDB(MenuBase):
    id: str
    user_id: str
    items: List[MenuItemInDB] = []
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class Menu(MenuInDB):
    pass 

File: frontend/src/types/menu.ts
export interface MenuItem {
  id: string;
  name: string;
  description?: string;
  price: number;
  category: string;
  menu_id: string;
}

export interface Menu {
  id: string;
  name: string;
  user_id: string;
  items: MenuItem[];
  created_at: string;
  updated_at: string;
}

export interface MenuCategory {
  id: string;
  name: string;
  items: MenuItem[];
}

export interface MenuItemCreate {
  name: string;
  description?: string;
  price: number;
  category: string;
}

export interface MenuItemUpdate {
  name?: string;
  description?: string;
  price?: number;
  category?: string;
}

export interface MenuCreate {
  name: string;
  items?: MenuItemCreate[];
}

export interface MenuUpdate {
  name?: string;
}
